Javatrix.java

```diff
@@ -25,6 +25,50 @@
     int n;
 
     /**
+     * Construct a matrix from a 2D array.
+     *
+     * @param arr - 2D array to be copied into matrix
+     **/
+    public Javatrix(double[][] arr)
+    {
+        this.m = arr.length;
+        this.n = arr[0].length;
+        matrix = new double[m][n];
+        for (int i = 0; i < m; i++)
+        {
+            for (int j = 0; j < n; j++)
+            {
+                matrix[i][j] = arr[i][j];
+            }
+        }
+    }
+
+    /**
+     * Construct an x by y matrix from a 2D array.
+     * 
+     * @param arr - 2D array to be copied into matrix
+     * @param x - number of rows
+     * @param y - number of columns
+     */
+    public Javatrix(double[][] arr, int x, int y)
+    {
+        this.m = x;
+        this.n = y;
+        matrix = new double[m][n];
+        for (int i = 0; i < arr.length; i++)
+        {
+            for (int j = 0; j < arr[0].length; j++)
+            {
+                // If index not out of range for arr.
+                if (i < m && j < n)
```

Comment:
The API suggests that arguments don't need to be checked. Do you think the loop should just test i < m and j < n and only copy those specified rows/cols from the provided array, versus attempting to copy the whole thing and having to test the arguments?